src/JsonSchema/Validator.php

```diff
@@ -19,37 +20,47 @@
  * @author Bruno Prieto Reis <bruno.p.reis@gmail.com>
  * @see    README.md
  */
-class Validator extends Constraint
+class Validator extends BaseConstraint
 {
     const SCHEMA_MEDIA_TYPE = 'application/schema+json';
 
     /**
      * Validates the given data against the schema and returns an object containing the results
      * Both the php object and the schema are supposed to be a result of a json_decode call.
-     * The validation works as defined by the schema proposal in http://json-schema.org
+     * The validation works as defined by the schema proposal in http://json-schema.org.
+     *
+     * Note that the first argument is passwd by reference, so you must pass in a variable.
      *
      * {@inheritDoc}
      */
-    public function check($value, $schema = null, JsonPointer $path = null, $i = null)
+    public function validate(&$value, $schema = null, $checkMode = null)
     {
+        $initialCheckMode = $this->factory->getConfig();
+        if ($checkMode !== null) {
```

Comment:
No. `validate()` only clobbers `checkMode` if the user actually passes it. If `$checkMode` is left as null, it uses whatever was already set on the factory. This is why the default value on this method is `null`.

The actual default *is* `CHECK_MODE_NORMAL` - it's set as the third argument to `Factory::__construct()`.